server/protocols/jmap/src/main/java/org/apache/james/jmap/VersionParser.java

```diff
@@ -47,37 +52,34 @@ public VersionParser(Set<Version> supportedVersions, JMAPConfiguration jmapConfi
         Preconditions.checkArgument(supportedVersions.contains(jmapConfiguration.getDefaultVersion()),
                 "%s is not a supported JMAP version", jmapConfiguration);
 
-        this.supportedVersions = supportedVersions;
+        this.supportedVersions = supportedVersions.stream()
+            .collect(Guavate.toImmutableMap(version -> version.asString().toLowerCase(Locale.US)));
```

Comment:
Why we convert Set to Map, and key still is `version.asString`?
is it only makes sense when `Version` object has more than one property?

Reply:
Lookup in a map is O(1) - globally, lookup in a set iterating elements is O(n).

Also initiating the pipeline is quite expensive - we do execute it on each JMAP request so quite often...